lib/parallel.js

```diff
@@ -3,31 +3,98 @@
 const qRouter = require('.');
 
 function createParallelRouter(spec) {
+    const pq = {...spec};
```

Comment:
The spread operator does a deep copy for "top-level" "not-nested" data, and then a shallow copy, memory reference for nested stuff.

The reason for the assignment in the first place is mainly for readability. Also to avoid assignments to a function parameter, again for readability.